db/gorm.go

```diff
@@ -142,6 +142,45 @@ type rows[T any] struct {
 	numCols int
 }
 
+// ForEach calls the given function with each model object retrieved by
+// repeatedly calling .Get(). It closes the rows object when it hits an error
+// or finishes iterating over the rows, so it can only be called once. This is
+// the intended way to use the result of QueryContext or QueryFrom; the other
+// methods on this type are lower-level and intended for advanced use only.
+func (r rows[T]) ForEach(do func(*T) error) (err error) {
+	defer func() {
+		// Close the row reader when we exit. Use the named error return to combine
```

Comment:
I remember you worked out this trick a while ago, but did you ever find it documented? I.e. is it guaranteed to be the case that if you `return foo`, that assigned to the named return values before executing any defer statements? That seems like really surprising behavior.

Reply:
[Yes](https://go.dev/ref/spec#Return_statements):
> A "return" statement that specifies results sets the result parameters before any deferred functions are executed.